src/app/Plans/components/Wizard/SelectVMs.tsx

```diff
@@ -0,0 +1,141 @@
+} from '@patternfly/react-table';
+import tableStyles from '@patternfly/react-styles/css/components/Table/table';
+
+import { IVMs } from '../../types';
+import { useSortState, usePaginationState, useSelectionState } from '@app/common/hooks';
+import { StatusIcon, StatusType } from '@konveyor/lib-ui';
+
+interface ISelectVMsProps {
+  VMs: IVMs[];
+}
+
+const SelectVMs: React.FunctionComponent<ISelectVMsProps> = ({ VMs }: ISelectVMsProps) => {
+  const getSortValues = (vm: IVMs) => {
+    return ['', vm.Name, vm.Datacenter, vm.Cluster, vm.Host, vm.FolderPath, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(VMs, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const { selectedItems, toggleItemSelected, areAllSelected, selectAll } = useSelectionState<IVMs>(
+    sortedItems
+  );
+
+  const { selectedItems: expandedVMs, toggleItemSelected: toggleVMExpanded } = useSelectionState<
+    IVMs
+  >(sortedItems);
+
+  const columns: ICell[] = [
+    {
+      // Using a custom column instead of Table's onSelect prop due to issues
+      title: (
+        <input
+          type="checkbox"
+          aria-label="Select all VMs"
+          onChange={(event: React.FormEvent<HTMLInputElement>) => {
+            selectAll(event.currentTarget.checked);
+          }}
+          checked={areAllSelected}
+        />
+      ),
+      columnTransforms: [classNamesTransform(tableStyles.tableCheck)],
+    },
+    {
+      title: 'Migration Analysis',
+      transforms: [sortable, wrappable],
+      cellTransforms: [compoundExpand],
+    },
+    { title: 'VM Name', transforms: [sortable, wrappable] },
+    { title: 'Datacenter', transforms: [sortable] },
+    { title: 'Cluster', transforms: [sortable] },
+    { title: 'Host', transforms: [sortable] },
+    { title: 'Folder Path', transforms: [sortable, wrappable] },
+  ];
+
+  const rows: IRow[] = [];
+
+  currentPageItems.forEach((vm: IVMs) => {
+    const isSelected = selectedItems.includes(vm);
+    const isExpanded = expandedVMs.includes(vm);
+    rows.push({
+      meta: { vm },
+      isOpen: isExpanded,
+      cells: [
+        {
+          title: (
+            <input
+              type="checkbox"
+              aria-label={`Select vm ${vm.Name}`}
+              onChange={(event: React.FormEvent<HTMLInputElement>) => {
+                toggleItemSelected(vm, event.currentTarget.checked);
+              }}
+              checked={isSelected}
+            />
+          ),
+        },
+        {
+          title: (
+            <StatusIcon status={StatusType[vm.MigrationAnalysis]} label={vm.MigrationAnalysis} />
+          ),
+        },
+        vm.Name,
+        vm.Datacenter,
+        vm.Cluster,
+        vm.Host,
+        vm.FolderPath,
+      ],
+    });
+    if (isExpanded) {
+      rows.push({
+        parent: 1,
+        compoundExpand: 1,
```

Comment:
Per @vconzola , we need to replace this compound expansion with regular expansion, which I think just means removing the `compoundExpand` property. See this example: https://www.patternfly.org/v4/documentation/react/components/table#compact-expandable (but obviously with our hooks approach instead of the class stuff)

Reply:
Sorry, disregard my comment here. That was just for the mappings tables.

Reply:
I Initially used an Expandable Table but that didn't work as I explained during scrum last week.
Effectively the (Compact) Expandable format the second row with the same number of columns which limits the width of the text we have to display.
So I thought about a Compound Expandable Table was better look & feel.

Reply:
Let's talk later so you can show me what you mean. You should be able to just use a colSpan to get around that.